pkg/controller/engine.go

```diff
@@ -147,14 +148,14 @@ func (e *Engine) done(name string, err error) {
 
 // Watch an object.
 type Watch struct {
-	kind       runtime.Object
+	kind       client.Object
```

Comment:
Is this change required due to an upstream change, or would `runtime.Object` still work here? I'd prefer to stick with `runtime.Object` if it's easy to do so, since we use it throughout our codebase.

Reply:
In regard to providers, what I've been briefly tried out, most of them are fine except for some tests which are defining `MockGet`. The signature of `MockGet` (and rest of `MockXXX` for that matter) has changed a providers need a bit of adjustment for this to compile properly on them.